### PR TITLE
RFC 176 doc updates for ember-metal package

### PR DESCRIPTION
**Pull request by karthiicksiva**
#15723

### CODE REVIEW

**Comment by karthiick** [comment]
@locks I am not sure about how to update the reference to ember-runtime package link. 

```
  At the heart of Ember is Ember-Runtime, a set of core functions that provide
  cross-platform compatibility and object property observing.  Ember-Runtime is
  small and performance-focused so you can use it alongside other
  cross-platform libraries such as jQuery. For more details, see
  [Ember-Runtime](https://emberjs.com/api/modules/ember-runtime.html).

```

https://github.com/emberjs/ember.js/blob/v2.18.0/packages/ember-metal/lib/core.js#L16

After 2.15 version of the docs, it seems ember-runtime removed (https://emberjs.com/api/ember/2.15/modules/ember-runtime). How to proceed?

File: packages/ember-metal/lib/map.js [hunk 1]
```diff
@@ -186,7 +186,7 @@ class OrderedSet {
 
   /**
     @method copy
-    @return {Ember.OrderedSet}
+    @return {OrderedSet}
     @private
   */
   copy() {
```
Comment (toddjordan) [inline packages/ember-metal/lib/map.js]: on line 220, we still have `@namespace` for Map.  We should remove this line so Map shows up as `Map` in the docs and not `Ember.Map`

File: packages/ember-metal/lib/map.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ class OrderedSet {
   /**
     @method create
     @static
-    @return {Ember.OrderedSet}
+    @return {OrderedSet}
```
Comment (toddjordan) [inline packages/ember-metal/lib/map.js]: According to https://github.com/ember-cli/ember-rfc176-data/blob/master/mappings.json its still `Ember.OrderedSet`

File: packages/ember-metal/lib/map.js [hunk 3]
```diff
@@ -63,7 +63,7 @@ class OrderedSet {
   /**
```
Comment (toddjordan) [inline packages/ember-metal/lib/map.js]: We need to make the `@module` .   `@ember/map` and not `Ember`

File: packages/ember-metal/lib/run_loop.js [hunk 4]
```diff
@@ -161,7 +161,7 @@ run.join = function() {
   });
   ```
 
-  In this example, we use Ember.run.bind to bind the setupEditor method to the
+  In this example, we use `@ember/runloop/bind` to bind the setupEditor method to the
```
Comment (toddjordan) [inline packages/ember-metal/lib/run_loop.js]: I think you can just use `bind`

File: packages/ember-metal/lib/watching.js [hunk 5]
```diff
@@ -21,7 +21,7 @@ import {
   invokes `Ember.propertyWillChange` and `Ember.propertyDidChange`. This is the
   primitive used by observers and dependent keys; usually you will never call
   this method directly but instead use higher level methods like
-  `Ember.addObserver()`
+  `@ember/object/observers/addObserver`
```
Comment (toddjordan) [inline packages/ember-metal/lib/watching.js]: Just `addObserver` should do.

File: packages/ember-metal/tests/accessors/get_path_test.js [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { moduleFor, AbstractTestCase } from 'internal-test-helpers';
 
 let obj;
 
-moduleFor('Ember.get with path', class extends AbstractTestCase {
+moduleFor('@ember/object/get with path', class extends AbstractTestCase {
```
Comment (toddjordan) [inline packages/ember-metal/tests/accessors/get_path_test.js]: `get with path`

File: packages/ember-metal/tests/accessors/get_properties_test.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import { getProperties } from '../..';
 import { moduleFor, AbstractTestCase } from 'internal-test-helpers';
 
-moduleFor('Ember.getProperties', class extends AbstractTestCase {
+moduleFor('@ember/object/getProperties', class extends AbstractTestCase {
```
Comment (toddjordan) [inline packages/ember-metal/tests/accessors/get_properties_test.js]: `getProperties`

File: packages/ember-metal/tests/is_blank_test.js [hunk 8]
```diff
@@ -1,8 +1,8 @@
 import { isBlank } from '..';
 
-QUnit.module('Ember.isBlank');
+QUnit.module('@ember/utils/isBlank');
```
Comment (toddjordan) [inline packages/ember-metal/tests/is_blank_test.js]: `isBlank`

File: packages/ember-metal/tests/is_empty_test.js [hunk 9]
```diff
@@ -4,9 +4,9 @@ import {
   OrderedSet
 } from '..';
 
-QUnit.module('Ember.isEmpty');
+QUnit.module('@ember/utils/isEmpty');
 
-QUnit.test('Ember.isEmpty', function(assert) {
+QUnit.test('@ember/utils/isEmpty', function(assert) {
```
Comment (toddjordan) [inline packages/ember-metal/tests/is_empty_test.js]: `isEmpty`

**Comment by toddjordan** [comment]
Generally, I'd like for functions like `isEmpty` to be referred to by name alone or module separate from name, like "`isEmpty` from '@ember/utils'". Using `@ember/utils/isEmpty` mixes path with function and is unclear/inaccurate.

**Comment by toddjordan** [comment]
@karthiick we should delete that block of documentation for `Ember`

**Comment by karthiick** [comment]
@toddjordan I have made the changes. Let me know, if i missed something.

File: packages/ember-metal/lib/map.js [hunk 10]
```diff
@@ -2,7 +2,7 @@ import { assert } from 'ember-debug';
 import { guidFor } from 'ember-utils';
 
 /**
-@module ember
+@module ember/map
```
Comment (toddjordan) [inline packages/ember-metal/lib/map.js]: Change to `@ember/map` and we are good

**Comment by toddjordan** [comment]
@karthiick this is great.  I commented on just one more change and then I will merge.

**Comment by karthiick** [comment]
@toddjordan Done!

**Comment by toddjordan** [comment]
Looks great! Will merge when the CI passes.  Thanks @karthiick !!

**Pull request merged**